### DISCUSSION TITLE
Git: Permission denied (password,publickey)

### DISCUSSION THREAD

**Discussion by aborruso** [WhitewaterFoundry/Pengwin#719] [Q&A] [answered]
Hi,
I have set my key using `ssh-keygen -C "user@domain.it"`, and I have edited my `~/.ssh/config` file in this way:

```
Host git.domain.it
 IdentityFile ~/.ssh/id_rsa
 User a.borruso
```

When I run the clone command, it seems that git uses as username my Linux username and not the one of `~/.ssh/id_rsa`. When I run

```
git clone ssh://git.domain.it/SBU-B2B/pkb115/_git/pkb115
```

I have

```
Cloning into 'pkb115'...
Permission denied, please try again.
Permission denied, please try again.
aborr@git.planetek.it: Permission denied (password,publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

As you can see the user is `aborr` (my Linux) username and not the one of the config (`a.borruso`).

How to force to read the right user? How to force git to use my `~/.ssh/id_rsa`?

**Comment by crramirez** [comment, marked as answer]
Hello @aborruso ,

First check if you are using the Linux version of the programs with:

```
command -v git
command -v ssh
````

Regards,
Carlos

**Reply by aborruso** [reply]
Thank you, you are right, it's for my config.

Solved

![image](https://user-images.githubusercontent.com/30607/141500227-7cd1a37e-9bf2-47cf-bbdd-3d2f4639d748.png)
